velox/functions/sparksql/ArrayRepeat.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+class RepeatFunction : public exec::VectorFunction {
+ public:
+  static constexpr int32_t kMaxResultEntries =
+      std::numeric_limits<int32_t>::max() - 15;
+
+  bool isDefaultNullBehavior() const override {
+    // repeat(null, n) returns an array of n nulls.
+    return false;
+  }
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    VectorPtr localResult;
+    if (args[1]->isConstantEncoding()) {
+      localResult = applyConstant(rows, args, outputType, context);
+      if (localResult == nullptr) {
+        return;
+      }
+    } else {
+      localResult = applyFlat(rows, args, outputType, context);
+    }
+    context.moveOrCopyResult(localResult, rows, result);
+  }
+
+ private:
+  VectorPtr applyConstant(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context) const {
+    const auto numRows = rows.end();
+    auto pool = context.pool();
+
+    auto* constantCount = args[1]->as<ConstantVector<int32_t>>();
+    if (constantCount->isNullAt(0)) {
+      // If count is a null constant, the result should be all nulls.
+      return BaseVector::createNullConstant(outputType, numRows, pool);
+    }
+
+    auto count = constantCount->valueAt(0);
+    // Negative count is like zero count, and empty array will be returned.
+    count = count < 0 ? 0 : count;
+    VELOX_USER_CHECK_LE(
+        count,
+        kMaxResultEntries,
+        "Count argument of array_repeat function must be less than or equal to 2147483632 (MAX_INT32 - 15)");
```

Comment:
drop '2147483632', it will be included in the error message automatically

than or equal to 'MAX_INT32 - 15'